### DISCUSSION TITLE
Could ex_cldr_plugs be folded into ex_cldr?

### DISCUSSION THREAD

**Discussion by petrus-jvrensburg** [elixir-cldr/cldr#211] [General]
I'm sure that everyone who uses _ex_cldr_ doesn't necessarily need _ex_cldr_plugs_, but it is such a light-weight package that it wouldn't do any harm just including it by default. It doesn't add any compile-time overhead, or download any CLDR data from what I can see.

The benefit of folding _ex_cldr_plugs_ into _ex_cldr_ would be to make it easier to get started with CLDR in a phoenix application: one less package to install, and one less thing to explain as a separate concept to new users.

Any thoughts?

**Comment by kipcole9** [comment]
I was interested to see if there were any other ex_cldr lib users that might have an opinion - but perhaps this isn't the best place to get broader feedback.

The package composition of the various libraries has been driven by two primary goals:

1. Limit memory consumption since each library embeds differing parts of the CLDR respository.
2. Allow different upgrade cycles (and bug fix releases) since the scope of CLDR is vast and having to update one package just because a small corner was updated seems to be to have more friction that desirable.

There have been many cases in the past where the ISO currency database has been updated - but that's the only change so only `ex_cldr_currencies` was affected. Similarly for `cldr_utils` where there was a period where there were several bug fix releases that didn't have any affect on the main libraries.

I agree there is a risk of thinking this whole ecosystem is too complex.  I think the better approach is to be clearer about what needs to be configured.  Because I started with `ex_cldr` there is an unreasonably focus on this being the "main" library. I reality its the "support act".

If the documentation focused primarily on configuring only `ex_cldr_numbers` (or units or calendars or lists or money or plugs) then it would look a lot simpler. And since all the required dependencies are configured transitively there would typically only be one configured library.

**Reply by petrus-jvrensburg** [reply]
Aah... yes I haven't thought about it that way.

Let me ask around to see if we can get a bit more feedback, and also think about how we could pitch it differently in the docs if that makes more sense.
